### ISSUE TITLE
More verbose error: Variable not found

### ISSUE THREAD

**Issue by sebastianbeyer** [alex-robinson/ncio#5]
`ncio:: error: NetCDF: Variable not found`

It would be nice to know, which variable causes this error.
I have looked into the source and it appears that this error message comes from the underlying netcdff library, so it should be fixed there, but maybe ncio could catch this error and provide additional information until then?

**Comment by sebastianbeyer** [comment]
Okai, this is more complicated than I thought...
What happens is that `nc4_read_internal_numeric` calls `nc_check (nf90_inq_varid(nc_id, trim(name), v_id))`. `nf90_inq_varid(nc_id, trim(name), v_id)` returns an errorcode, which is handled by `nc_check` (it translates it to the message `Variable not found` with help of `nf90_strerror` and then stops execution). 
The problem now is, that `nc_check` does not know anything about the missing variable that caused the error and can not report on this.

**Comment by perrette** [comment]
I made a modif in nc_read to make a more informative error message. Can you check if that works for you? 
8d53fe6923d8bfb197f3c00064282d6de801284f

**Comment by sebastianbeyer** [comment]
I am thinking about something like a `weak_nc_check` that does not stop execution, but returns the error to the calling procedure. The calling procedure could then generate the full errormessage with all information. 
I have to think about this a little bit more to get something more general...

**Comment by perrette** [comment]
Actually, this is already the role of `nc_check` to ease error handling in general. A `status` variable is returned by the netcdf method that does what you suggest. What about the solution I implemented?

**Comment by sebastianbeyer** [comment]
Oh, you answered already, that looks nice! I'll give it a try!

**Comment by sebastianbeyer** [comment]
yeah! That's what I wanted! Thank you!

**Comment by sebastianbeyer** [comment]
The problem I have with `nc_check` is, that it stops execution, so there is no way to get any more information. The solution you gave with splitting calling `nc_check` and giving some more information is nice and easy, I feel like it is a little bit 'hacky' if you know what I mean?

**Comment by alex-robinson** [comment]
Mahé's solution looks pretty good to me. The problem is that the native netcdf error is poorly done, so inevitably our solution will probably not be ideal...

**Comment by perrette** [comment]
Hey, can you check out 675da34ea66f999f327b5f61305a6e2648f2815c
You should be able to add an optional iostat parameter for exception catching.
(if that works, the branch could be merged)
Note you can also use `nc_exists_var`,  `nc_exists_attr_variable` and  `nc_exists_attr_global`.

**Comment by alex-robinson** [comment]
Looks fine to me, did you test it?

**Comment by sebastianbeyer** [comment]
Yes, I like it a lot! But I don't know if it still fits your idea of a lean and minimal, easy to use interface? Also it returns the netcdf- (not ncio) errorcode, so mixes some things up... so maybe the first solution of Mahe, is better for the general library?

**Comment by alex-robinson** [comment]
It's fine with me. The argument is optional, so in principle it can be
considered for "advanced users" without complicating the basic use of the
routines. For me, I think Mahé's solution is enough - normally when I get
this error, it's because I typed a filename wrong or something. But I could
see how being able to manage the error yourself is useful.

Mahé, you want to push it?

**Comment by perrette** [comment]
Yo I merged it.

I did not check the issue pointed by @sebastianbeyer, that there might be an inconsistency with other functions. Need to be checked. And maybe decide for a simpler return code of 0 and -1 (e.g. integer, parameter :: NC_SUCCESS = 0, NC_ERROR = -1) to consistently use across the module.

Also need to expand that approach across the module, methods (at least nc_read_attr methods).

I reopen the issue for now (though actually we deviated quite a bit from the original issue...).
